### PR TITLE
client/tailscale: document ServeConfig accessors a bit more

### PR DESCRIPTION
**Pull request by bradfitz**
Updates tailscale/corp#7515

### CODE REVIEW

File: client/tailscale/localclient_test.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2021 Tailscale Inc & AUTHORS All rights reserved.
```
Comment (DentonGentry) [inline client/tailscale/localclient_test.go:1]: 2022

**Review by DentonGentry — APPROVED** [review]

File: client/tailscale/localclient.go [hunk 2]
```diff
@@ -862,7 +862,7 @@ func (lc *LocalClient) NetworkLockSign(ctx context.Context, nodeKey key.NodePubl
 // SetServeConfig sets or replaces the serving settings.
 // If config is nil, settings are cleared and serving is disabled.
 func (lc *LocalClient) SetServeConfig(ctx context.Context, config *ipn.ServeConfig) error {
-	b, err := json.Marshal(&config)
+	b, err := json.Marshal(config)
```
Comment (shayne) [inline client/tailscale/localclient.go:865]: 🤦

**Review by shayne — APPROVED** [review]

File: client/tailscale/localclient_test.go (hunk 1 above)
Comment (bradfitz) [reply client/tailscale/localclient_test.go]: done

File: client/tailscale/localclient.go (hunk 2 above)
Comment (bradfitz) [reply client/tailscale/localclient.go:865]: got a fun one coming up for ya

**Pull request merged**